### PR TITLE
[FIX] payment_razorpay: fix duplicate token creation issue

### PR DESCRIPTION
**Pull request by kcv-odoo**
Steps:
- Install sale and razopay app.
- Set-up razorpay provider.
- Create quotation.
- Pay that quotation via razorpay and save payment details.
- Duplicate that quote and pay it via already saved method.

Issue:
- Creating duplicate token when paying via token.

Cause:
- Since [PR] we are creating token in odoo when razorpay give token_id in notification_data and tokenization is enabled on razorpay and in some case when we pay quote via token it return token in notification_data and since we forgot to check if transaction is already connected to token or not it always create new token :(

Fix:
- Add condition to check if transaction is connected to token or not and only create token if transaction is not connected to token. Additionally pass entity_data in token creation method so it pass proper data when we don't find `id` in `notification_data`.

[PR]: https://github.com/odoo/odoo/pull/159250

### CODE REVIEW

File: addons/payment_razorpay/models/payment_transaction.py [hunk 1]
```diff
@@ -405,8 +405,9 @@ def _process_notification_data(self, notification_data):
             if self.provider_id.capture_manually:
                 self._set_authorized()
         elif entity_status in const.PAYMENT_STATUS_MAPPING['done']:
-            if entity_data.get('token_id') and self.provider_id.allow_tokenization:
-                self._razorpay_tokenize_from_notification_data(notification_data)
+            if not self.token_id and entity_data.get('token_id') \
+               and self.provider_id.allow_tokenization:
```
Comment (AntoineVDV) [inline addons/payment_razorpay/models/payment_transaction.py:409]:
```suggestion
            if (
                not self.token_id
                and entity_data.get('token_id')
                and self.provider_id.allow_tokenization
            ):
```

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

Edit: This could use a test ;)

File: addons/payment_razorpay/models/payment_transaction.py (hunk 1 above)
Comment (kcv-odoo) [reply addons/payment_razorpay/models/payment_transaction.py]: Done! Could you check the test case? :see_no_evil:

**Review by AntoineVDV — APPROVED** [review]
Hello @kcv-odoo, I force-pushed the branch to replace the `search_count` by a `mock.call_count` and took the opportunity to clean a bit the other tests ;)

@robodoo r+

**Pull request closed without merging**